runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/FileSystemWatcher.java

```diff
@@ -0,0 +1,274 @@
+package io.aklivity.zilla.runtime.engine;
```

Comment:
Few things we need to decide with this:

- Where should it belong? Create a separate dir for this?
- It has some common parts with the ConfigWatcher, so after both is merged (or before this is merged), we can do a little cleanup. Maybe create a FileWatcherHelper?
- Currently, we're always creating the FileSystemWatcher, even if filesystem binding is never used. We can implement it in a way, that it's only created once if we need to watch for a file. Maybe when we're adding a FileSystemBinding, we can create it?

Reply:
Suggest keeping this `FileSystemWatcher` in `binding-filesystem` until we have a design that could support reuse by both engine config watcher and filesystem binding.

The main issue is that moving it up to `DispatchAgent` should really make `DispatchAgent` responsible for calling `.take()`, because if multiple bindings were to do that it would get very confusing as they encountered each others watch events. Then if `DispatchAgent` is calling `.take()` (or non-blocking equivalent) then it needs to dispatch those watch events via some clean abstractions.

We do have something similar for network I/O via `Poller` and `PollerKey`, which is included in the `doWork` loop of `DispatchAgent`.